src/database/DataTypes/StocktakeBatch.js

```diff
@@ -211,7 +211,25 @@ export class StocktakeBatch extends Realm.Object {
    */
   set countedTotalQuantity(quantity) {
     // Handle packsize of 0.
-    this.countedNumberOfPacks = this.packSize ? quantity / this.packSize : 0;
+    const countedNumberOfPacks = this.packSize ? quantity / this.packSize : 0;
+
+    this.countedNumberOfPacks = countedNumberOfPacks;
+    this.doses = countedNumberOfPacks * this.dosesPerVial;
+  }
+
+  get isVaccine() {
+    return this.itemBatch?.item?.isVaccine ?? false;
+  }
+
+  get dosesPerVial() {
+    return this.isVaccine ? this.itemBatch?.item?.doses ?? 0 : 0;
```

Comment:
Should this be:

```suggestion
    return this.isVaccine ? this.itemBatch?.doses ?? 0 : 0;
```

Also, think you can make it a bit more concise (but possibly a bit less explicit):

```suggestion
    return this.isVaccine && this.itemBatch?.doses || 0;
```

Reply:
your suggestion needs to have parenteitheris `(this.isVaccine && this.itemBatch?.doses) || 0`

But nah, need to use `item.doses` which is the number of doses per vial as above